pkg/processors/recover/kubernetes/port_in_use_recover.go

```diff
@@ -0,0 +1,96 @@
+package kubernetes
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os/exec"
+
+	"github.com/go-logr/logr"
+
+	"github.com/kubediag/kubediag/pkg/processors"
+	"github.com/kubediag/kubediag/pkg/processors/diagnoser/kubernetes"
+	"github.com/kubediag/kubediag/pkg/processors/utils"
+)
+
+const (
+	ContextKeyPortInUseRecoverResult = "recover.kubernetes.port_in_use.result"
+)
+
+// portInUseRecover recover the bug happens when port is in use (connection reset by peer).
+type portInUseRecover struct {
```

Comment:
A diagnoser is sufficient for such scenario. The Recoverer to restart any service could be implemented as a general case in another pull request. Use `Recoverer` instead of `Recover` in naming.

Reply:
Have deleted the Recoverer in this pull request.
etc. To use ```Recoverer``` instead of ```Recover``` in naming, current processor ```subpath_remount_recover``` and folder /processors/recover (and related docs) need to be renamed too.